#!/bin/bash

# sample-c1
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T HaplotypeCaller -R input_files/ref_genome/hg19.fa -I input_files/dcm_bam/Control1_RG_MD_IR_BQ.bam --genotyping_mode DISCOVERY -stand_emit_conf 10 -stand_call_conf 30 -o input_files/dcm_vcf_haplotypeCaller/c1_raw_variants.vcf
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T VariantRecalibrator -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/c1_raw_variants.vcf -resource:hapmap,known=false,training=true,truth=true,prior=15.0 input_files/gatk_variantRecali/hapmap_3.3.hg19.sites.vcf -resource:omni,known=false,training=true,truth=false,prior=12.0 input_files/gatk_variantRecali/1000G_omni2.5.hg19.sites.vcf -resource:1000G,known=false,training=true,truth=false,prior=10.0 input_files/gatk_variantRecali/1000G_phase1.snps.high_confidence.hg19.sites.vcf -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 input_files/dbsnp/dbsnp_138.hg19.vcf -an DP -an QD -an FS -an SOR -an MQRankSum -an ReadPosRankSum -mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 -recalFile input_files/dcm_vcf_haplotypeCaller/c1_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/c1_recalibrate_SNP.tranches
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T ApplyRecalibration -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/c1_raw_variants.vcf -mode SNP --ts_filter_level 99.0 -recalFile input_files/dcm_vcf_haplotypeCaller/c1_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/c1_recalibrate_SNP.tranches -o c1_recalibrated_snps_raw_indels.vcf

# sample-c2
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T HaplotypeCaller -R input_files/ref_genome/hg19.fa -I input_files/dcm_bam/Control2_RG_MD_IR_BQ.bam --genotyping_mode DISCOVERY -stand_emit_conf 10 -stand_call_conf 30 -o input_files/dcm_vcf_haplotypeCaller/c2_raw_variants.vcf
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T VariantRecalibrator -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/c2_raw_variants.vcf -resource:hapmap,known=false,training=true,truth=true,prior=15.0 input_files/gatk_variantRecali/hapmap_3.3.hg19.sites.vcf -resource:omni,known=false,training=true,truth=false,prior=12.0 input_files/gatk_variantRecali/1000G_omni2.5.hg19.sites.vcf -resource:1000G,known=false,training=true,truth=false,prior=10.0 input_files/gatk_variantRecali/1000G_phase1.snps.high_confidence.hg19.sites.vcf -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 input_files/dbsnp/dbsnp_138.hg19.vcf -an DP -an QD -an FS -an SOR -an MQRankSum -an ReadPosRankSum -mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 -recalFile input_files/dcm_vcf_haplotypeCaller/c2_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/c2_recalibrate_SNP.tranches
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T ApplyRecalibration -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/c2_raw_variants.vcf -mode SNP --ts_filter_level 99.0 -recalFile input_files/dcm_vcf_haplotypeCaller/c2_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/c2_recalibrate_SNP.tranches -o c2_recalibrated_snps_raw_indels.vcf

# sample-p1
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T HaplotypeCaller -R input_files/ref_genome/hg19.fa -I input_files/dcm_bam/Patient1_RG_MD_IR_BQ.bam --genotyping_mode DISCOVERY -stand_emit_conf 10 -stand_call_conf 30 -o input_files/dcm_vcf_haplotypeCaller/p1_raw_variants.vcf
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T VariantRecalibrator -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/p1_raw_variants.vcf -resource:hapmap,known=false,training=true,truth=true,prior=15.0 input_files/gatk_variantRecali/hapmap_3.3.hg19.sites.vcf -resource:omni,known=false,training=true,truth=false,prior=12.0 input_files/gatk_variantRecali/1000G_omni2.5.hg19.sites.vcf -resource:1000G,known=false,training=true,truth=false,prior=10.0 input_files/gatk_variantRecali/1000G_phase1.snps.high_confidence.hg19.sites.vcf -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 input_files/dbsnp/dbsnp_138.hg19.vcf -an DP -an QD -an FS -an SOR -an MQRankSum -an ReadPosRankSum -mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 -recalFile input_files/dcm_vcf_haplotypeCaller/p1_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/p1_recalibrate_SNP.tranches
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T ApplyRecalibration -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/p1_raw_variants.vcf -mode SNP --ts_filter_level 99.0 -recalFile input_files/dcm_vcf_haplotypeCaller/p1_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/p1_recalibrate_SNP.tranches -o p1_recalibrated_snps_raw_indels.vcf

# sample-p2
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T HaplotypeCaller -R input_files/ref_genome/hg19.fa -I input_files/dcm_bam/Patient2_RG_MD_IR_BQ.bam --genotyping_mode DISCOVERY -stand_emit_conf 10 -stand_call_conf 30 -o input_files/dcm_vcf_haplotypeCaller/p2_raw_variants.vcf
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T VariantRecalibrator -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/p2_raw_variants.vcf -resource:hapmap,known=false,training=true,truth=true,prior=15.0 input_files/gatk_variantRecali/hapmap_3.3.hg19.sites.vcf -resource:omni,known=false,training=true,truth=false,prior=12.0 input_files/gatk_variantRecali/1000G_omni2.5.hg19.sites.vcf -resource:1000G,known=false,training=true,truth=false,prior=10.0 input_files/gatk_variantRecali/1000G_phase1.snps.high_confidence.hg19.sites.vcf -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 input_files/dbsnp/dbsnp_138.hg19.vcf -an DP -an QD -an FS -an SOR -an MQRankSum -an ReadPosRankSum -mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 -recalFile input_files/dcm_vcf_haplotypeCaller/p2_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/p2_recalibrate_SNP.tranches
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T ApplyRecalibration -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/p2_raw_variants.vcf -mode SNP --ts_filter_level 99.0 -recalFile input_files/dcm_vcf_haplotypeCaller/p2_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/p2_recalibrate_SNP.tranches -o p2_recalibrated_snps_raw_indels.vcf

# sample-p3
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T HaplotypeCaller -R input_files/ref_genome/hg19.fa -I input_files/dcm_bam/Patient3_RG_MD_IR_BQ.bam --genotyping_mode DISCOVERY -stand_emit_conf 10 -stand_call_conf 30 -o input_files/dcm_vcf_haplotypeCaller/p3_raw_variants.vcf
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T VariantRecalibrator -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/p3_raw_variants.vcf -resource:hapmap,known=false,training=true,truth=true,prior=15.0 input_files/gatk_variantRecali/hapmap_3.3.hg19.sites.vcf -resource:omni,known=false,training=true,truth=false,prior=12.0 input_files/gatk_variantRecali/1000G_omni2.5.hg19.sites.vcf -resource:1000G,known=false,training=true,truth=false,prior=10.0 input_files/gatk_variantRecali/1000G_phase1.snps.high_confidence.hg19.sites.vcf -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 input_files/dbsnp/dbsnp_138.hg19.vcf -an DP -an QD -an FS -an SOR -an MQRankSum -an ReadPosRankSum -mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 -recalFile input_files/dcm_vcf_haplotypeCaller/p3_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/p3_recalibrate_SNP.tranches
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T ApplyRecalibration -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/p3_raw_variants.vcf -mode SNP --ts_filter_level 99.0 -recalFile input_files/dcm_vcf_haplotypeCaller/p3_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/p3_recalibrate_SNP.tranches -o p3_recalibrated_snps_raw_indels.vcf

# sample-p4
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T HaplotypeCaller -R input_files/ref_genome/hg19.fa -I input_files/dcm_bam/Patient4_RG_MD_IR_BQ.bam --genotyping_mode DISCOVERY -stand_emit_conf 10 -stand_call_conf 30 -o input_files/dcm_vcf_haplotypeCaller/p4_raw_variants.vcf
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T VariantRecalibrator -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/p4_raw_variants.vcf -resource:hapmap,known=false,training=true,truth=true,prior=15.0 input_files/gatk_variantRecali/hapmap_3.3.hg19.sites.vcf -resource:omni,known=false,training=true,truth=false,prior=12.0 input_files/gatk_variantRecali/1000G_omni2.5.hg19.sites.vcf -resource:1000G,known=false,training=true,truth=false,prior=10.0 input_files/gatk_variantRecali/1000G_phase1.snps.high_confidence.hg19.sites.vcf -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 input_files/dbsnp/dbsnp_138.hg19.vcf -an DP -an QD -an FS -an SOR -an MQRankSum -an ReadPosRankSum -mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 -recalFile input_files/dcm_vcf_haplotypeCaller/p4_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/p4_recalibrate_SNP.tranches
java -Xmx4g -jar lib/GenomeAnalysisTK.jar -T ApplyRecalibration -R input_files/ref_genome/hg19.fa -input input_files/dcm_vcf_haplotypeCaller/p4_raw_variants.vcf -mode SNP --ts_filter_level 99.0 -recalFile input_files/dcm_vcf_haplotypeCaller/p4_recalibrate_SNP.recal -tranchesFile input_files/dcm_vcf_haplotypeCaller/p4_recalibrate_SNP.tranches -o p4_recalibrated_snps_raw_indels.vcf
